### PR TITLE
Development

### LINKED ISSUE #251: More data classes
Looking forward, we have to get rid of Map<key, another map> in key parts of the codebase and move to data classes. This will improve performance and code maintainability

### LINKED ISSUE #260: Multi-Profil-Unterstützung
Aus Google Play übernommen:

_Wünschenswert (Feature-Request) wäre noch Multi-Profil-Unterstützung, sodass man als "Eltern" auch mal die Zugangsdaten "mehrerer" Kinder hinterlegen kann um schnell zwischen diesen "Profilen"/Accounts hin und her switchen zu können._

Das wäre durchaus interessant, würde aber auch ein grundsätzliches Refactoring der Speicherstruktur bedeuten. (Das wollte ich sowieso mal machen, wenn keine PRs da sind. (Also das Refactoring)

### LINKED ISSUE #282: Refactor Applet support and launch structure
To further support more applets (also from Teacher and Parent side) a structural redesign of the core app has to be done in order to support different views (eg "mein Unterricht" as teacher). 
Also the logout functionality is kind of broken due to the methods of applet validation currently in place.

### PR DESCRIPTION
**Pull request by alessioC42**
closes #251 
closes #260 
closes #282
#288 in some ways

### CODE REVIEW

**Comment by alessioC42** [comment]
@kurwjan , I am pretty sure, that I broke parts of the Notification system with the new CombinedAppletBuilder, but since I do not use it I also do not really know how it should behave. Would be great if you looked into that and fixed the applet. 

I removed the current filtering logic on behalf of creating a new one or integrating the existing system differently later before the launch of 3.0.0

Background services are also currently not fully implemented. This PR sets a new base other PRs can build uppon. For example the notification feature for the "Messages" or "Lessons" applet.

File: app/lib/view/settings/subsettings/notifications.dart [hunk 1]
```diff
@@ -35,114 +39,152 @@ class NotificationElements extends StatefulWidget {
...
+                  _notificationPermissionStatus = _notificationPermissionStatus;
+                });
+              },
+              trailing:
+                  (_notificationPermissionStatus == PermissionStatus.granted)
+                      ? Icon(Icons.check, color: Colors.green)
+                      : Icon(Icons.error, color: Colors.red),
+            ),
+            const Divider(),
+            FutureBuilder<int>(
```
Comment (kurwjan) [inline app/lib/view/settings/subsettings/notifications.dart:118]: Add forgotten check

**Review by kurwjan — CHANGES_REQUESTED** [review]
Some small things:
image
+ Colors are in small case and there's a missing divider between dynamic color and app colors.
image
+ There's an abundant divider on the top (looks weird to me). Also the timetable applet doesn't have a selected icon.
+ In dark mode the account switcher button is a bit too dark
+  Add account button should be replaced with a FAB
+ In notifications green tick should be a primary color tick
+ Maybe add a divider between "Alle Anwesenheiten" and the rest, currently it's only more elevated than the rest which is difficult to see in dark mode.
+ Yep conversations isn't correctly working anymore (not updating stuff like setting the unread alert)

The rest is fantastic. The current project organisation and database is much better. I love the new offline applet selector (I need to investigate it more) and the CombinedAppletBuilder.

**Review by Rajala1404 — APPROVED** [review]
Besides the Back button (that i already fixed) and the translation of Exception causes i have nothing to complain

**Review by kurwjan — APPROVED** [review]
Ich aktzeptiere das mal bevor ich es zu lang ziehe

**Pull request merged**